frontend/app/src/components/history/ExternalTradeForm.vue

```diff
@@ -340,36 +330,71 @@ const { assetSymbol } = useAssetInfoRetrieval();
 const baseSymbol = assetSymbol(base);
 const quoteSymbol = assetSymbol(quote);
 
-const baseRules = [
-  (v: string) => !!v || t('external_trade_form.validation.non_empty_base')
-];
-const quoteRules = [
-  (v: string) => !!v || t('external_trade_form.validation.non_empty_quote')
-];
-const amountRules = [
-  (v: string) => !!v || t('external_trade_form.validation.non_empty_amount')
-];
-const rateRules = [
-  (v: string) => !!v || t('external_trade_form.validation.non_empty_rate')
-];
-const quoteAmountRules = [
-  (v: string) =>
-    !!v || t('external_trade_form.validation.non_empty_quote_amount')
-];
-
-const feeRules = [
-  (v: string) =>
-    !!v ||
-    !get(feeCurrency) ||
-    t('external_trade_form.validation.non_empty_fee')
-];
-
-const feeCurrencyRules = [
-  (v: string) =>
-    !!v ||
-    !get(fee) ||
-    t('external_trade_form.validation.non_empty_fee_currency')
-];
+const rules = {
+  base: {
+    required: helpers.withMessage(
+      t('external_trade_form.validation.non_empty_base').toString(),
+      required
+    )
+  },
+  quote: {
+    required: helpers.withMessage(
+      t('external_trade_form.validation.non_empty_quote').toString(),
+      required
+    )
+  },
+  amount: {
+    required: helpers.withMessage(
+      t('external_trade_form.validation.non_empty_amount').toString(),
+      required
+    )
+  },
+  rate: {
+    required: helpers.withMessage(
+      t('external_trade_form.validation.non_empty_rate').toString(),
+      required
+    )
+  },
+  quoteAmount: {
+    required: helpers.withMessage(
+      t('external_trade_form.validation.non_empty_quote_amount').toString(),
+      required
+    )
+  },
+  fee: {
+    required: helpers.withMessage(
+      t('external_trade_form.validation.non_empty_fee').toString(),
+      requiredIf(() => !!get(feeCurrency))
```

Comment:
can't we just pass the feeCurrency ref directly to requiredIf? 

Reply:
we can, but I have do some type conversion 
``` feeCurrency as unknown as Ref<boolean> ```